Microsoft.Toolkit.Uwp.UI/Converters/FormatStringConverter.cs

```diff
@@ -0,0 +1,57 @@
+﻿// ******************************************************************
+// Copyright (c) Microsoft. All rights reserved.
+// This code is licensed under the MIT License (MIT).
+// THE CODE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+using System;
+using Windows.UI.Xaml.Data;
+
+namespace Microsoft.Toolkit.Uwp.UI.Converters
+{
+    /// <summary>
+    /// Value converter that converts an <see cref="IFormattable"/> to a formatted <see cref="string"/>.
+    /// The string format needs to be passed as the converter parameter.
+    /// </summary>
+    public class FormatStringConverter : IValueConverter
+    {
+        /// <summary>
+        /// Convert an <see cref="IFormattable"/> value to a formatted <see cref="string"/>.
+        /// </summary>
+        /// <param name="value">The source data being passed to the target.</param>
+        /// <param name="targetType">The type of the target property, as a type reference.</param>
+        /// <param name="parameter">The format string.</param>
+        /// <param name="language">The language of the conversion. Not used.</param>
+        /// <returns>The formatted string.</returns>
+        public object Convert(object value, Type targetType, object parameter, string language)
+        {
+            var formattableValue = value as IFormattable;
+
+            if (formattableValue == null)
+            {
+                return value;
+            }
+
+            return formattableValue.ToString(parameter as string, null);
```

Comment:
If parameter is null or not a string, will this explode? If so, we should probably either return the value if parameter isn't a string, or have it throw an ArgumentException... I'd vote for returning the value (same as if formattableValue == null).

    if (formattableValue == null || !(parameter is string))